v2/tools/generator/internal/astmodel/local_package_reference.go

```diff
@@ -87,18 +87,26 @@ func (pr LocalPackageReference) String() string {
 // We don't check the version prefix (which contains the version of the generator) as that may contain alpha or beta
 // even if the ARM version is not preview.
 func (pr LocalPackageReference) IsPreview() bool {
-	return containsPreviewVersionLabel(strings.ToLower(pr.version))
+	return containsPreviewVersionLabel(strings.ToLower(pr.apiVersion))
 }
 
 // WithVersionPrefix returns a new LocalPackageReference with a different version prefix
 func (pr LocalPackageReference) WithVersionPrefix(prefix string) LocalPackageReference {
-	pr.versionPrefix = prefix
+	pr.generatorVersion = prefix
 	return pr
 }
 
 // HasVersionPrefix returns true if we have the specified version prefix, false otherwise.
 func (pr LocalPackageReference) HasVersionPrefix(prefix string) bool {
-	return pr.versionPrefix == prefix
+	return pr.generatorVersion == prefix
+}
+
+func (pr LocalPackageReference) GeneratorVersion() string {
+	return pr.generatorVersion
+}
+
+func (pr LocalPackageReference) ApiVersion() string {
```

Comment:
minor: [this](https://github.com/golang/go/wiki/CodeReviewComments#initialisms) suggests that we should say `APIVersion`, not `ApiVersion`.

I think we violate this rule in places (I know I've fixed a few that I named that way), but should try to honor it going forward IMO